### PR TITLE
Restoring build scripts back to using Roslyn.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* After removing the WCF dependency on the old version of MSBuild via Mono which was not supported in Roslyn we can now remove the temporary hack we had in place that allowed us to continue using the old version of MSBuild.

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
The Innerloop tests on Linux should validate this change.
@iamjasonp and @roncain feel free to merge if the checks pass and you are satisfied everything works on Linux.

File: build.sh [hunk 1]
```diff
@@ -140,11 +109,8 @@ build_managed_corefx()
...
-    # $__scriptpath/Tools/corerun $__scriptpath/Tools/MSBuild.exe "$__buildproj" /nologo /verbosity:minimal "/fileloggerparameters:Verbosity=normal;LogFile=$__buildlog" "/l:BinClashLogger,$__binclashloggerdll;LogFile=$__binclashlog" /t:Build /p:ConfigurationGroup=$__BuildType /p:OSGroup=$__BuildOS /p:COMPUTERNAME=$(hostname) /p:USERNAME=$(id -un) /p:TestNugetRuntimeId=$__TestNugetRuntimeId $__UnprocessedBuildArgs
+	
+	$__scriptpath/Tools/corerun $__scriptpath/Tools/MSBuild.exe "$__buildproj" /nologo /verbosity:minimal "/fileloggerparameters:Verbosity=normal;LogFile=$__buildlog" "/l:BinClashLogger,$__binclashloggerdll;LogFile=$__binclashlog" /t:Build /p:ConfigurationGroup=$__BuildType /p:OSGroup=$__BuildOS /p:COMPUTERNAME=$(hostname) /p:USERNAME=$(id -un) /p:TestNugetRuntimeId=$__TestNugetRuntimeId $__UnprocessedBuildArgs
```
Comment (iamjasonp) [inline build.sh]: NIT: tabs to spaces

**Comment by iamjasonp** [comment]
I think you have to remove `/override.targets` - this is causing a redirect to mono.targets which doesn't exist anymore and is causing this failure - but this causes a bunch of other issues for me, like:

```
/home/jason/git/wcf-test/Tools/Microsoft.Common.CurrentVersion.targets(1097,5): warning MSB3644: The reference assemblies for framework ".NETPortable,Version=v4.5,Profile=Profile7" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [/home/jason/git/wcf-test/src/System.Private.ServiceModel/tests/Common/Infrastructure/src/Infrastructure.Common.csproj]
/home/jason/git/wcf-test/Tools/Microsoft.Common.CurrentVersion.targets(1097,5): warning MSB3644: The reference assemblies for framework ".NETPortable,Version=v4.5" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [/home/jason/git/wcf-test/src/System.Private.ServiceModel/tests/Common/Infrastructure/src/Infrastructure.Common.csproj]
```

**Comment by roncain** [comment]
I tried manually commenting out override.targets and see the failures Jason saw.

File: build.sh (hunk 1 above)
Comment (StephenBonikowsky) [inline build.sh]: Fixed.

**Comment by StephenBonikowsky** [comment]
Removed override.targets and add code from corefx to dir.targets that addresses the new errors.

File: dir.targets [hunk 2]
```diff
@@ -18,6 +18,18 @@
   <!-- permit a wrapping build system to contribute targets to this build -->
   <Import Condition="Exists('$(MSBuildThisFileDirectory)..\open.targets')" Project="$(MSBuildThisFileDirectory)..\open.targets" />
 
+  <PropertyGroup>
+    <!-- We don't use any of MSBuild's resolution logic for resolving the framework, so just set these two properties to any folder that exists to skip
```
Comment (roncain) [inline dir.targets]: There is a small chance of TFS build issues when we touch these files.  So we either merge and monitor TFS rolling builds, or I can buddy-build this in TFS and report back in a couple hours.

**Comment by roncain** [comment]
An attempted build in Linux with the new PR had the same issues, but clearly CI passed.  I'd like to spend a few minutes investigating my own Linux failures.  It could be due to the difference in the way CI is calling build.sh

**Pull request merged**